### PR TITLE
doc: add developper guidelines

### PR DESCRIPTION
**Pull request by outscale-jju**
Signed-off-by: Jerome Jutteau jerome.jutteau@outscale.com

### CODE REVIEW

File: doc/contrib.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+make bench
+```
+And to run a specific benchmark (still from your build directory):
+```
+./tests/pmtud/bench.sh
+```
+
+# Developping a new brick
+
+## First things to think about
```
Comment (outscale-mgo) [inline doc/contrib.md]: "First step" may be a better name

File: doc/contrib.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Maybe you will have some question to ask yourself before developping a new
+brick:
+- Is the brick I would like to create only do the job related to it's job ?
```
Comment (outscale-mgo) [inline doc/contrib.md]: I'm not sure our brick are really kiss(the firewall is everything but kiss), nor peoples will care about this, if someone need a brick he will just create and that's all.

File: doc/contrib.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+Try not to add too much features. Stay [KISS](https://en.wikipedia.org/wiki/KISS_principle).
+- What is my brick interface ? That the first thing you may write: add a new
+file in e.g. `include/packetgraph/newbrick.h` and write your interface by
+looking at others to see how it's done. Comments your functions so you are
+really clear on which interface you would like to propose. Don't import
+external libs from public API (all files in `include/packetgraph/*`).
+- Internally, bricks can be MONOPOLE meaning that they can only be connected
```
Comment (outscale-mgo) [inline doc/contrib.md]: *which mean the the brick can have only one neighbor .

File: doc/contrib.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+on one of their side. That the case with nic bricks or vhost bricks. Bricks can
+be DIPOLE: they can be connected only once on each side. Finally, we have
```
Comment (outscale-mgo) [inline doc/contrib.md]: *only once per side

File: doc/contrib.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+- Add your brick in `/README.md` file
+- You should provide the most important tests by creating a new folder in
+`tests` folder.
+- Add benchmarks with your tests (if possible)
+- Add a test checking that brick cannot send an empty burst
+- If your test can be run using travis (test that does not need to start qemu),
+then add it to `/.travis.yml`
+- Bonus: add a very simple example of your brick in `/examples` folder with
+very limited external dependencies
+- Bonus: use valgrind when running your test to check any potential memory
```
Comment (outscale-mgo) [inline doc/contrib.md]: not really a bonus this one

File: README.md [hunk 6]
```diff
@@ -105,6 +105,11 @@ $ make install
 Packetgraph project is published under [GNU GPLv3](http://www.gnu.org/licenses/quick-guide-gplv3.en.html).
 For more information, check LICENSE file.
 
+# Contribute
+
+New to packetgraph ? Want to contribute and/or create a new brick ?
```
Comment (outscale-mgo) [inline README.md]: we should give a link to linux coding style too, as we it: https://www.kernel.org/doc/Documentation/CodingStyle

**Comment by outscale-mgo** [comment]
merge
